src/codesyncd/handlers/buffer_handler.ts

```
@@ -223,20 +231,36 @@ export class bufferHandler {
 		}
 
 		if (!canConnect) return CodeSyncState.set(CODESYNC_STATES.BUFFER_HANDLER_RUNNING, false);
-
 		CodeSyncState.set(CODESYNC_STATES.BUFFER_HANDLER_RUNNING, true);
-		
+
 		try {
 			// Check if we have an active user
 			this.activeUser = new UserState().getUser();
 			if (!this.activeUser) return CodeSyncState.set(CODESYNC_STATES.BUFFER_HANDLER_RUNNING, false);
 			const diffs = await this.getDiffFiles();
-			if (!diffs.files.length) return CodeSyncState.set(CODESYNC_STATES.BUFFER_HANDLER_RUNNING, false);
-			if (canSendDiffs) CodeSyncLogger.debug(`Processing ${diffs.files.length}/${diffs.count} diffs, uuid=${this.instanceUUID}`);
-			const repoDiffs = this.groupRepoDiffs(diffs.files);
+			// Get tabs data
+			const tabsHandler = new TabsHandler();
+			const tabYMLFiles = await tabsHandler.getYMLFiles();
+			let tabsData: ITabYML[] = [];
+			let repoDiffs: IRepoDiffs[] = [];
+
+			if (canSendSocketData) {
+				if (tabYMLFiles.files.length > 0) {
+					CodeSyncLogger.debug(`Processing ${tabYMLFiles.files.length}/${tabYMLFiles.count} tabs, uuid=${this.instanceUUID}`);
+					tabsData = tabsHandler.getTabsData(tabYMLFiles.files);
+				}
+
+				if (diffs.files.length > 0) {
+					CodeSyncLogger.debug(`Processing ${diffs.files.length}/${diffs.count} diffs, uuid=${this.instanceUUID}`);
+					repoDiffs = this.groupRepoDiffs(diffs.files);
+				}
+			}
+
+			if (!tabYMLFiles.files.length && !diffs.files.length) return CodeSyncState.set(CODESYNC_STATES.BUFFER_HANDLER_RUNNING, false);;
```

This can be moved to line 246 (Fail Fast)